Add commands for setting provider scope on env resource

bjoginapally opened the pull request:
Description
- Add rad env update command to populate provider scope on env resource
- Add flag to clear provider scope in the same command
issue: https://github.com/project-radius/radius/issues/5089

rad env update default --azure-subscription-id testSubId --azure-resource-group testResourceGroup
<img width="590" alt="Screenshot 2023-03-09 at 2 16 55 AM" src="https://user-images.githubusercontent.com/99223262/223849275-2f84b12a-2470-4cab-a02e-85f4aa8a5eea.png">

rad env update default --azure-subscription-id testSubId
<img width="1255" alt="Screenshot 2023-03-09 at 2 35 17 AM" src="https://user-images.githubusercontent.com/99223262/223849777-4bdd5f7a-d4cd-478f-9ee9-3392736d8d37.png">

rad env update default --azure-subscription-id testSubId --azure-resource-group testResourceGroup --clear-azure
<img width="1257" alt="Screenshot 2023-03-09 at 2 46 21 AM" src="https://user-images.githubusercontent.com/99223262/223852118-4a328c6e-32dc-4419-9a01-c6003ac928eb.png">



rad env update default --clear-azure --clear-aws
<img width="516" alt="Screenshot 2023-03-09 at 2 42 32 AM" src="https://user-images.githubusercontent.com/99223262/223851279-7db3628f-d3dd-4e28-91d7-08a82df4c7c5.png">






## Issue reference

<!--
We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.
-->

Fixes: radius-project/core-team#1100 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Adds necessary unit tests for change
* [ ] Adds necessary E2E tests for change
* [x] Unit tests passing
* [ ] Extended the documentation / Created issue for it

Commit:
Add commands for setting provider scope on env resource
- Add rad env update command to populate provider scope on env resource
- Add flag to clear provider scope in the same command
issue: https://github.com/project-radius/radius/issues/5089

Commit:
Address rev-1 comments

ytimocin reviewed (approved):
Need more documentation. Other than that **LGTM**. I would also get another thumbs up from someone else too.

Commit:
Address comments on rev-2

Commit:
Fix clear providers

Commit:
Fix bugs
- Fix setting of provider irrepective of user provider input
- Fix overwriting of previous provider irrespective of user input
- Command prints help message when no flags are defined along

AaronCrawfis reviewed (approved):
Played around with it a bunch and looks good!

I am able to inject a '/' character into any field and break the scope of the environment, but that seems just as bad an outcome as putting in a non-existent subscription or resource group name. I wonder what the error case would be for that. But doesn't seem major
